Interesting addition to the project! I tried to install it, but got the following error:

```shell
koan@tux:~/decoder$ pip3 install scikit-build
Collecting scikit-build
  Downloading scikit_build-0.12.0-py2.py3-none-any.whl (73 kB)
     |████████████████████████████████| 73 kB 981 kB/s 
Requirement already satisfied: packaging in /home/koan/.local/lib/python3.8/site-packages (from scikit-build) (20.4)
Requirement already satisfied: distro in /home/koan/.local/lib/python3.8/site-packages (from scikit-build) (1.5.0)
Requirement already satisfied: wheel>=0.29.0 in /home/koan/.local/lib/python3.8/site-packages (from scikit-build) (0.36.2)
Requirement already satisfied: setuptools>=28.0.0; python_version >= "3" in /home/koan/.local/lib/python3.8/site-packages (from scikit-build) (56.0.0)
Requirement already satisfied: six in /usr/lib/python3/dist-packages (from packaging->scikit-build) (1.14.0)
Requirement already satisfied: pyparsing>=2.0.2 in /home/koan/.local/lib/python3.8/site-packages (from packaging->scikit-build) (2.4.7)
Installing collected packages: scikit-build
Successfully installed scikit-build-0.12.0
koan@tux:~/decoder$ cd python/
koan@tux:~/decoder/python$ pip3 install .
Processing /home/koan/decoder/python
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
Building wheels for collected packages: TheengsDecoder
  Building wheel for TheengsDecoder (PEP 517) ... error
  ERROR: Command errored out with exit status 1:
   command: /usr/bin/python3 /tmp/tmpabptxyqf build_wheel /tmp/tmpw6ibufvx
       cwd: /tmp/pip-req-build-n8pexc0s
  Complete output (88 lines):
  Not searching for unused variables given on the command line.
  -- The C compiler identification is GNU 9.3.0
  -- Detecting C compiler ABI info
  -- Detecting C compiler ABI info - done
  -- Check for working C compiler: /usr/bin/cc - skipped
  -- Detecting C compile features
  -- Detecting C compile features - done
  -- The CXX compiler identification is GNU 9.3.0
  -- Detecting CXX compiler ABI info
  -- Detecting CXX compiler ABI info - done
  -- Check for working CXX compiler: /usr/bin/c++ - skipped
  -- Detecting CXX compile features
  -- Detecting CXX compile features - done
  -- Configuring done
  -- Generating done
  -- Build files have been written to: /tmp/pip-req-build-n8pexc0s/_cmake_test_compile/build
  -- The C compiler identification is GNU 9.3.0
  -- The CXX compiler identification is GNU 9.3.0
  -- Detecting C compiler ABI info
  -- Detecting C compiler ABI info - done
  -- Check for working C compiler: /usr/bin/cc - skipped
  -- Detecting C compile features
  -- Detecting C compile features - done
  -- Detecting CXX compiler ABI info
  -- Detecting CXX compiler ABI info - done
  -- Check for working CXX compiler: /usr/bin/c++ - skipped
  -- Detecting CXX compile features
  -- Detecting CXX compile features - done
  -- Found PythonInterp: /usr/bin/python3 (found version "3.8.10")
  -- Found PythonLibs: /usr/lib/x86_64-linux-gnu/libpython3.8.so (found version "3.8.10")
  -- Performing Test Weak Link MODULE -> SHARED (gnu_ld_ignore) - Success
  _modinit_prefix:PyInit_
  -- Configuring done
  CMake Error at CMakeLists.txt:6 (add_library):
    Cannot find source file:
  
      ../src/decoder.cpp
  
    Tried extensions .c .C .c++ .cc .cpp .cxx .cu .mpp .m .M .mm .ixx .cppm .h
    .hh .h++ .hm .hpp .hxx .in .txx .f .F .for .f77 .f90 .f95 .f03 .hip .ispc
  
  
  CMake Error at CMakeLists.txt:6 (add_library):
    No SOURCES given to target: _decoder
  
  
  CMake Generate step failed.  Build files cannot be regenerated correctly.
    File "/tmp/pip-build-env-xs6aub0x/overlay/lib/python3.8/site-packages/skbuild/setuptools_wrap.py", line 586, in setup
      env = cmkr.configure(cmake_args,
    File "/tmp/pip-build-env-xs6aub0x/overlay/lib/python3.8/site-packages/skbuild/cmaker.py", line 237, in configure
      raise SKBuildError(
  
  
  --------------------------------------------------------------------------------
  -- Trying "Ninja" generator
  --------------------------------
  ---------------------------
  ----------------------
  -----------------
  ------------
  -------
  --
  --
  -------
  ------------
  -----------------
  ----------------------
  ---------------------------
  --------------------------------
  -- Trying "Ninja" generator - success
  --------------------------------------------------------------------------------
  
  Configuring Project
    Working directory:
      /tmp/pip-req-build-n8pexc0s/_skbuild/linux-x86_64-3.8/cmake-build
    Command:
      cmake /tmp/pip-req-build-n8pexc0s -G Ninja -DCMAKE_INSTALL_PREFIX:PATH=/tmp/pip-req-build-n8pexc0s/_skbuild/linux-x86_64-3.8/cmake-install -DPYTHON_EXECUTABLE:FILEPATH=/usr/bin/python3 -DPYTHON_VERSION_STRING:STRING=3.8.10 -DPYTHON_INCLUDE_DIR:PATH=/usr/include/python3.8 -DPYTHON_LIBRARY:FILEPATH=/usr/lib/x86_64-linux-gnu/libpython3.8.so -DSKBUILD:INTERNAL=TRUE -DCMAKE_MODULE_PATH:PATH=/tmp/pip-build-env-xs6aub0x/overlay/lib/python3.8/site-packages/skbuild/resources/cmake -DCMAKE_BUILD_TYPE:STRING=Release
  
  Traceback (most recent call last):
  
  An error occurred while configuring with CMake.
    Command:
      cmake /tmp/pip-req-build-n8pexc0s -G Ninja -DCMAKE_INSTALL_PREFIX:PATH=/tmp/pip-req-build-n8pexc0s/_skbuild/linux-x86_64-3.8/cmake-install -DPYTHON_EXECUTABLE:FILEPATH=/usr/bin/python3 -DPYTHON_VERSION_STRING:STRING=3.8.10 -DPYTHON_INCLUDE_DIR:PATH=/usr/include/python3.8 -DPYTHON_LIBRARY:FILEPATH=/usr/lib/x86_64-linux-gnu/libpython3.8.so -DSKBUILD:INTERNAL=TRUE -DCMAKE_MODULE_PATH:PATH=/tmp/pip-build-env-xs6aub0x/overlay/lib/python3.8/site-packages/skbuild/resources/cmake -DCMAKE_BUILD_TYPE:STRING=Release
    Source directory:
      /tmp/pip-req-build-n8pexc0s
    Working directory:
      /tmp/pip-req-build-n8pexc0s/_skbuild/linux-x86_64-3.8/cmake-build
  Please see CMake's output for more information.
  ----------------------------------------
  ERROR: Failed building wheel for TheengsDecoder
Failed to build TheengsDecoder
ERROR: Could not build wheels for TheengsDecoder which use PEP 517 and cannot be installed directly
```

Do you have any idea what's wrong? This is on Ubuntu 20.04.

--- a/python/README.md
+++ b/python/README.md
@@ -11,7 +11,7 @@ apt-get install cmake
 
 From a terminal cd to this folder and execute:
 ```
-pip install .
+python setup.py install --user
 ```
 
 ## using

